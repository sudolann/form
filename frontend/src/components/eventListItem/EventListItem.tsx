import React, { FunctionComponent, ReactElement } from 'react';
import { EventProps } from '../eventById/EventById';

export const EventListItem: FunctionComponent<EventProps> = ({
  name,
  email,
  date,
}): ReactElement => {
  return (
    <li className="list--li">
      <p>name: {name}</p>
      <p>email: {email}</p>
      <p>date: {date}</p>
    </li>
  );
};
