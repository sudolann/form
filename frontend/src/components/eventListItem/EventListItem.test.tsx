import React from 'react';
import { EventListItem } from './EventListItem';
import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';

describe('EventListItem', () => {
  it('should take a snapshot', async () => {
    const { asFragment } = render(
      <EventListItem
        name="test"
        email="me@email.com"
        date="20.01.2021"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders props', () => {
    const { getByText } = render(
      <EventListItem
        name="test"
        email="me@email.com"
        date="20.01.2021"
      />,
    );
    const eventName = getByText(/name: test/i);
    expect(eventName).toBeInTheDocument();
    const email = getByText(/email: me@email.com/i);
    expect(email).toBeInTheDocument();
    const date = getByText(/date: 20.01.2021/i);
    expect(date).toBeInTheDocument();
  });
});
