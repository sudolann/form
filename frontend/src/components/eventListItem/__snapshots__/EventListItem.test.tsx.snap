// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListItem should take a snapshot 1`] = `
<DocumentFragment>
  <li
    class="list--li"
  >
    <p>
      name: test
    </p>
    <p>
      email: me@email.com
    </p>
    <p>
      date: 20.01.2021
    </p>
  </li>
</DocumentFragment>
`;
